src/coffea/dataset_tools/dataset_query.py

```diff
@@ -0,0 +1,366 @@
+        table.row_styles = ["dim", "none"]
+        Nfiles = len(outfiles)
+
+        sorted_sites = dict(
+            sorted(sites_counts.items(), key=lambda x: x[1], reverse=True)
+        )
+        for i, (site, stat) in enumerate(sorted_sites.items()):
+            table.add_row(str(i), site, f"{stat} / {Nfiles}", f"{stat*100/Nfiles:.1f}%")
+
+        self.console.print(table)
+        strategy = Prompt.ask(
+            "Select sites",
+            choices=["round-robin", "choice", "quit"],
+            default="round-robin",
+        )
+
+        files_by_site = defaultdict(list)
+
+        if strategy == "choice":
+            ind = list(
+                map(int, Prompt.ask("Enter list of sites index to be used").split(" "))
+            )
+            sites_to_use = [list(sorted_sites.keys())[i] for i in ind]
+            print(f"Filtering replicas with [green]: {' '.join(sites_to_use)}")
+
+            output = []
+            for ifile, (files, sites) in enumerate(zip(outfiles, outsites)):
+                random.shuffle(sites_to_use)
+                found = False
+                # loop on shuffled selected sites until one is found
+                for site in sites_to_use:
+                    try:
+                        iS = sites.index(site)
+                        output.append(files[iS])
+                        files_by_site[sites[iS]].append(files[iS])
+                        found = True
+                        break  # keep only one replica
+                    except ValueError:
+                        # if the index is not found just go to the next site
+                        pass
+
+                if not found:
+                    print(
+                        f"[bold red]No replica found compatible with sites selection for file #{ifile}. The available sites are:"
+                    )
+                    for f, s in zip(files, sites):
+                        print(f"\t- [green]{s} [cyan]{f}")
+                    return
+
+            self.replica_results[dataset] = output
+
+        elif strategy == "round-robin":
+            output = []
+            for ifile, (files, sites) in enumerate(zip(outfiles, outsites)):
+                # selecting randomly from the sites
+                iS = random.randint(0, len(sites) - 1)
+                output.append(files[iS])
+                files_by_site[sites[iS]].append(files[iS])
+            self.replica_results[dataset] = output
+
+        elif strategy == "quit":
+            print("[orange]Doing nothing...")
+            return
+
+        self.replica_results_bysite[dataset] = files_by_site
+
+        # Now let's print the results
+        tree = Tree(label=f"[bold orange]Replicas for [green]{dataset}")
+        for site, files in files_by_site.items():
+            T = tree.add(f"[green]{site}")
+            for f in files:
+                T.add(f"[cyan]{f}")
+        self.console.print(tree)
+
+    def do_whitelist_sites(self, args):
```

Comment:
Blocklist/allowlist, deny/allow, exempt/allowlist or block/permit all are good synonyms
(https://www.ietf.org/archive/id/draft-knodel-terminology-09.html)

Reply:
Also, docs should be clear about the precedence block-then-allow or allow-then-block

Reply:
@valsdav